superbit_lensing/shear_profiles/annular_jmac.py

```diff
@@ -176,24 +453,65 @@ def compute_profile(self, outfile, overwrite=False):
             gtan_err[i] = np.std(self.gtan[annulus]) / np.sqrt(n)
             gcross_err[i] = np.std(self.gcross[annulus]) / np.sqrt(n)
 
-            print("%.3f   %d   %f   %f   %f   %f" %\
-                  (midpoint_r[i], n, gtan_mean[i], gtan_err[i], gcross_mean[i], gcross_err[i])
-                  )
-
             i += 1
 
         table = Table()
         table.add_columns(
             [counts, midpoint_r, gtan_mean, gcross_mean, gtan_err, gcross_err],
             names=[
-                'counts', 'midpoint_r', 'gtan_mean', 'gcross_mean', 'gtan_err', 'gcross_err'
+                'counts', 'midpoint_r', 'mean_gtan', 'mean_gcross', 'err_gtan', 'err_gcross'
                 ],
             )
 
+
+        # Repeat calculation if an nfw table is supplied, and also compute shear bias
+        # Separating out the NFW loop significantly slows the code...
+        if nfw_tab is not None:
+            nfw_mid_r = np.zeros(N)
+            nfw_gtan_mean = np.zeros(N)
+            nfw_gtan_err = np.zeros(N)
+            nfw_gcross_mean = np.zeros(N)
+            nfw_gcross_err = np.zeros(N)
+
+            i = 0
+            for b1, b2 in zip(bins[:-1], bins[1:]):
+
+                annulus = (nfw_tab['r'] >= b1) & (nfw_tab['r'] < b2)
+                n = counts[i]
+                nfw_mid_r[i] = np.mean([b1, b2])
+                nfw_gtan_mean[i] = np.mean(nfw_tab['gtan'][annulus])
+                nfw_gcross_mean[i] = np.mean(nfw_tab['gcross'][annulus])
+                nfw_gtan_err[i] = np.std(nfw_tab['gtan'][annulus]) / np.sqrt(n)
+                nfw_gcross_err[i] = np.std(nfw_tab['gcross'][annulus]) / np.sqrt(n)
+
+                i += 1
+
+            # NFW shear profile should have same number of elements as galaxy shear profile
+            assert(len(nfw_mid_r) == len(midpoint_r))
+
+            table.add_columns(
+                [nfw_gtan_mean, nfw_gcross_mean, nfw_gtan_err, nfw_gcross_err],
+                names=['mean_nfw_gtan', 'mean_nfw_gcross', 'err_nfw_gtan', 'err_nfw_gcross'],
+                )
+
+            # Compute shear bias relative to reference NFW
+            compute_shear_bias(profile_tab=table)
+
         table.write(outfile, format='fits', overwrite=overwrite)
 
+        # Print to stdio for quickcheck -- open to better formatting if it exists
+        cols = table.colnames
+        colstring = '    '.join([col for col in cols])
+        print(colstring)
+
+        for i,e in enumerate(table):
+            print(f'{e.as_void()}')
+
+
         return
 
+
+
     def plot_profile(self, cat_file, truth_file, plot_file):
```

Comment:
This function will need a little work to be consistent w/ `master` now that #55 has been merged. I think what this will ultimately mean is for you to pull from `master` so you have my updates (basically getting rid of the comments so that the `ShearProfilePlotter` is actually run), and then have `profile_tab` passed to the plotter to also render the true NFW line. I think we could do the update to `shear_plots.py` in a different PR though.

Reply:
This seems like a good plan. I will make sure that `pipe_test` runs and we can assign this to an issue.